bitnami/logstash/templates/svc.yaml

```
@@ -17,6 +17,9 @@ spec:
   {{- if and (eq .Values.service.type "ClusterIP") .Values.service.clusterIP }}
   clusterIP: {{ .Values.service.clusterIP }}
   {{- end }}
+  {{- if (or (eq .Values.service.type "LoadBalancer") (eq .Values.service.type "NodePort")) }}
+  externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
+  {{- end }}
```

The current changes forces `externalTrafficPolicy` to be configured explicitly when using `LoadBalancer` or `NodePort`.

Kubernetes will automatically configure this to `Cluster` by default when missing, so could you please make this setting optional?

That way we will also prevent future issues with this chart if Kubernetes changes its default value.

```suggestion
  {{- if and .Values.service.externalTrafficPolicy (or (eq .Values.service.type "LoadBalancer") (eq .Values.service.type "NodePort")) }}
  externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
  {{- end }}
```

---

@migruiz4 I'm a bit confused. If I'm following you recommendation, this chart will not be aligned with other chart that has this setting already. 
If I keep the default value to `externalTrafficPolicy: Cluster` in values.yaml it's the same behaviour plus it's aligned with other charts.

---

Hi @ikus060,

You are right, almost all of the charts including the `externalTrafficPolicy` parameter have it configured as `externalTrafficPolicy: Cluster` by default.

Nonetheless, there is no standard for the condition, and I think the `and .Values.service.externalTrafficPolicy` condition to only set the value when the variable has been configured would make a good addition as it would prevent misconfigurations.

I have created an internal task to revisit and discuss this topic in the future and try to align a common configuration for all the charts.